thrift.js

```diff
@@ -57,18 +58,28 @@ var messageExceptionTypesDef = require('./message').exceptionTypesDef;
 
 var validThriftIdentifierRE = /^[a-zA-Z_][a-zA-Z0-9_\.]+$/;
 
-function Thrift(options) {
-    assert(options, 'options required');
-    assert(typeof options === 'object', 'options must be object');
-    assert(options.source || options.entryPoint, 'opts.entryPoint required');
+function Thrift(options, loadedCb) {
+    try {
+        assert(options, 'options required');
+        assert(typeof options === 'object', 'options must be object');
+        assert(options.source || options.entryPoint, 'opts.entryPoint required');
+    } catch (err) {
+        return this._exitOnError(loadedCb, err);
+    }
 
     // Coerce weakly-deprecated single include usage
     if (options.source) {
-        assert(typeof options.source === 'string', 'source must be string');
+        try {
+            assert(typeof options.source === 'string', 'source must be string');
+        } catch (err) {
+            return this._exitOnError(loadedCb, err);
+        }
```

Comment:
I think the comment was aimed at some other try/catch block initially. This should no longer be relevant thanks to the separation of the synchronous and asynchronous code paths to avoid Zalgo, as recommended.